app/(main)/league/[leagueId]/entry/[entryId]/week/Week.tsx

```diff
@@ -135,6 +138,35 @@ const Week = ({ entry, league, NFLTeams, week }: IWeekProps): JSX.Element => {
     } catch (error) {
       console.error('Could not load week data:', error);
       setError('Could not load week data.');
+    }
+  };
+
+  /**
+   * Fetches all entries for the current user.
+   * @returns {Promise<void>}
+   */
+  const getPickHistory = async (): Promise<void> => {
+    const entryId: string = entry;
+
+    try {
+      const entries = await getCurrentUserEntries(user.id, league);
+      const currentEntry = entries.find((entry) => entry.$id === entryId);
+
+      if (!currentEntry) {
+        throw new Error('Entry not found');
+      }
+
+      let entryHistory = currentEntry?.selectedTeams || [];
+
+      if (currentEntry?.selectedTeams.length > 0) {
+        entryHistory = entryHistory.map((teamName) =>
+          getNFLTeamLogo(NFLTeams, teamName),
+        );
+      }
+
+      setPickHistory(entryHistory);
+    } catch (error) {
+      throw new Error("Error fetching user's pick history");
```

Comment:
How come we're not adding a `console.error` here?

Reply:
well .. this is an application error. so if no entry is found then there are bigger issues. I am open to adding console.log instead of throw. Just don't know the reason for either/or? @shashilo ?